### PR TITLE
database: Add `RecordEncrypter` type

### PR DESCRIPTION
**Pull request by efritz**
This PR adds a type that encrypts batches of records matching a standardized but configurable form (à la dbworker). This utility is currently unused.

## Test plan

Unit tests in this PR.

### CODE REVIEW

File: internal/database/encryption.go [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+func NewRecordEncrypter(db DB) *RecordEncrypter {
+	return &RecordEncrypter{
+		Store: basestore.NewWithHandle(db.Handle()),
+	}
+}
+
+func (s *RecordEncrypter) Count(ctx context.Context, config EncryptionConfig) (numEncrypted int, numUnencrypted int, _ error) {
+	countQuery := sqlf.Sprintf(`
+		SELECT
+			(SELECT COUNT(*) FROM %s WHERE %s != '') AS encrypted,
```
Comment (eseliger) [inline internal/database/encryption.go]: heartbroken over the fact that it's `''` and not `null` 😢

File: internal/database/encryption.go [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+		return 0, nil
+	}
+
+	tx, err := s.Transact(ctx)
+	if err != nil {
+		return 0, err
+	}
+	defer func() { err = tx.Done(err) }()
+
+	values, err := config.Scan(tx.Query(ctx, sqlf.Sprintf(
```
Comment (eseliger) [inline internal/database/encryption.go:54]: not sure how fast encryption backends are but could this cause bad contention on a table with the for update and doing bulk processing?

File: internal/database/encryption_tables.go [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+	TableName           string
+	IDFieldName         string
+	KeyIDFieldName      string
+	EncryptedFieldNames []string
+	Scan                func(*sql.Rows, error) (map[int]Encrypted, error)
+	Key                 func() encryption.Key
+	Limit               int
+}
+
+var EncryptionConfigs = []EncryptionConfig{
```
Comment (eseliger) [inline internal/database/encryption_tables.go:22]: love how expressive this is

**Review by eseliger — APPROVED** [review]

File: internal/database/encryption_tables.go (hunk 3 above)
Comment (efritz) [reply internal/database/encryption_tables.go:22]: image

File: internal/database/encryption.go (hunk 2 above)
Comment (efritz) [reply internal/database/encryption.go:54]: If so then that table can lower their own limit as defined in the `EncryptionConfigs` (_encryption configs_ ^ `encryption configs`...).

File: internal/database/encryption.go (hunk 1 above)
Comment (efritz) [reply internal/database/encryption.go]: Backwards compatibility m8 🥲

**Pull request merged**